splash-reward-distributor/src/engine/queue.rs

```diff
@@ -142,11 +190,23 @@ where
                     let strike_time = <u64>::from_be_bytes(strike_time_bytes);
                     if strike_time <= current_time {
                         let task_id = &key[8..];
+                        let mut done_tasks =
+                            tx.iterator_cf_opt(tables.done, ReadOptions::default(), IteratorMode::Start);
+
+                        // We don't know the TX hash associated with `task_id`, so need to iterate
+                        // over the `done` column
+                        let task_is_done = done_tasks.any(|done_key| {
+                            if let Ok((done_key, _)) = done_key {
+                                *task_id == done_key[DONE_TASK_ID_IX_START..]
+                            } else {
+                                false
+                            }
+                        });
```

Comment:
It's not efficient to do this for each task, please avoid linear search unless you are 100% sure the collection you're searching through is always small.